### ISSUE TITLE
Support for browser wallets (like Alby)

### ISSUE THREAD

**Issue by reneaaron** [chill117/passport-lnurl-auth#6]
Currently this authentication strategy does not work with in-browser wallets like Alby. (see getAlby/lightning-browser-extension/issues/662 for details)

After debugging this strategy for quite some time I finally found the root cause, where `resave` is set to `true` when configuring the session.

```javascript
app.use(session({
	secret: '12345',
	resave: true, 
	saveUninitialized: true,
}));
```

While this works fine when you use a separate wallet (like on your mobile phone), using an in-browser wallet won't work because both the browser and the wallet share the same session and interfere with each other as `resave` will cause the changes on the session object to be overwritten at the end of the request. 

My quick-fix was to set `resave`to `false` but since there could be people out there who rely on this configuration it would probably be a good idea to change the implementation to somehow support both configurations.

**Comment by chill117** [comment]
I think you are referring to the [sample web server example](https://github.com/chill117/passport-lnurl-auth/blob/master/examples/simple.js). That is provided merely as a starting point for users of this module who would like to see a full working example. Nodejs and specifically express.js give a lot of freedom to the developer to decide how to build and structure their web applications. I don't think any changes are required to this module, unless I am missing something else?

**Comment by reneaaron** [comment]
While this issue was found in the sample web server example this module currently does not work with in-browser wallets for all applications that set their session configuration to `resave: true`. 

However, some applications won't be able to change their session configuration. I think it would be a nice thing if this module would support this use-case regardless of the specific session configuration in place. 

Please let me know if I can be of any help. 🙌

**Comment by chill117** [comment]
Fixed by 987fc5f9744ec00aea8d39cce7df90ac412f7e2d

It was necessary to overwrite the `req.session` object so that when "resave" is true, it includes the changes made to the session object. @reneaaron Thanks for reporting the issue and for providing more context - it helped to finally solve the problem.

**Comment by reneaaron** [comment]
Awesome, thanks for fixing it! 👍
